### PR TITLE
doc: account for PG autoscaling being the default

### PR DESCRIPTION
**Pull request by bitfehler**
The current documentation tries really hard to convince people to set `osd pool default pg_num` and `osd pool default pgp_num` in their configs, but at least the latter can be considered harmful on a fresh install of a version that has autoscaling enabled by default (at least quincy and beyond).

An example: you set up a new cluster, carefully calculate `pg_num` and `pgp_num`, and set those in your config to lets say `64`. When starting `radosgw`, it will fail. Why? Radosgw tries to create various pools without providing values for `pg_num` or `pgp_num`. This triggers the following in `OSDMonitor::prepare_new_pool()`:

- `pg_num` is set to `1`, because autoscaling is enabled
- `pgp_num` is set to `osd pool default pgp_num`, which we set to `64`
- This is an invalid setup, so the pool creation fails

Likewise, `ceph osd pool create mypool` (without providing values for `pg_num` or `pgp_num`) does not work.

Debugging this was essentially impossible without reading the source code, so this commit tries to at least improve the documentation. Besides updating the descriptions, it changes various provided config examples using the following rationale:

- Notably, everything at least works if a default value for neither `pg_num` nor `pgp_num` is provided
- If a pool is created without providing `pgp_num` explicitly, it will default to `osd pool default pgp_num` or, in absence of that, to the value of `pg_num`
- The latter is always what you want, unless you use advanced features, in which case you can be expected to set both values on pool creation
- Setting `osd pool default pgp_num` in your config breaks pool creation for various cases:
  - Pool creation without explicit values if autoscaling is enabled, which is the default (!)
  - Pool creation with explicit value for `pg_num`, but not `pgp_num`, if the configured value does not work for the provided `pg_num`
  - Pool creation with explicit value for `pg_num`, but not `pgp_num`, even if the configured value would work, but autoscaling is enabled
- On the other hand, not setting `osd pool default pgp_num` will always lead to `pgp_num = pg_num` unless explicitly specified
- Hence, I consider setting `osd pool default pgp_num` harmful in current versions of Ceph

I also took the liberty to put some more emphasis on the fact that the journal size is really only needed for filestore, which most new-comers are likely not using.

There is also a ticket which I think is very relevant and fixed by this, though it only captures part of the broader issue I am trying to prevent here:

Fixes: https://tracker.ceph.com/issues/47176

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by zdover23** [comment]
> I understand that this commit message probably violates your guideline of
> 
> > Additional comments MAY follow a blank line after the summary, but should be terse.
> 
> I find the context important for this PR and don't see why it shouldn't be preserved in the git history, but if you do so desire I am happy to shorten it.

@bitfehler: I personally have no problem with such a long commit message. I'll have to wait on technical approval from Neha or her designated proxy before I merge it and backport it, though.

File: doc/install/manual-deployment.rst
Comment (anthonyeleven) [inline doc/install/manual-deployment.rst:241]: See above

File: doc/rados/configuration/demo-ceph.conf
Comment (anthonyeleven) [inline doc/rados/configuration/demo-ceph.conf:24]: 👍

File: doc/rados/configuration/pool-pg.conf
Comment (anthonyeleven) [inline doc/rados/configuration/pool-pg.conf:19]: The first `3` above should be a `three`, would be groovy to amend that in since you're already in this file.

File: doc/rados/configuration/pool-pg.conf
Comment (anthonyeleven) [inline doc/rados/configuration/pool-pg.conf:15]: I think disabling this here is counter to the party line.

File: doc/rados/operations/pools.rst
Comment (anthonyeleven) [inline doc/rados/operations/pools.rst:61]: Suggest

>If you are not using the PG autoscaler you , may wish to explicitly set a value for :confval:`osd_pool_default_pg_num`, as the default is small and not ideal for many production-scale deployments.  Be careful, though, to not set a very high value as auto-deployed pools, notably certain RGW pools, will not hold much data and thus should not have a gratuitous number of PGs.  When the PG autoscaler is not actively managing placement group numbers, best practice is to explicitly provide `pg_num` and `pgp_num` when creating each pool.

One thing to consider here is that Ceph can deploy pools that don't contain much data and for which a `pg_num` in the hundreds or thousands may be drastic overkill, notably the panoply of pools that RGW uses.

File: doc/rados/operations/pools.rst
Comment (anthonyeleven) [inline doc/rados/operations/pools.rst:98]: Hardcoding defaults is discouraged, especially since they can change between releases.  Please replace this with something like

>The default value of :confval:`pg_num` is likely too small for production pools used for bulk data, including RBD and RGW data and bucket pools respectively.

Also, while you're here, on lines 94 and 105 please s/-/_/

**Review by anthonyeleven — CHANGES_REQUESTED** [review]

File: doc/install/manual-deployment.rst
Comment (bitfehler) [reply doc/install/manual-deployment.rst:355]: This whole section is about creating a filestore volume. Shall I remove the entire section?

File: doc/rados/operations/pools.rst
Comment (markhpc) [reply doc/rados/operations/pools.rst:61]: Indeed.  I would very much be in favor of merging some of these small pools together.  There's no reason to maintain so many separate pools for RGW.  In the mean time though, it's important to have good behavior in the data and bucket index pools primarily.  The rest are less important (imho).

File: doc/install/manual-deployment.rst
Comment (bitfehler) [reply doc/install/manual-deployment.rst:223]: Brought back the `pg_num` and replaced spaces with underscores in all files I touched, hope I didn't miss anything.

File: doc/rados/configuration/pool-pg.conf
Comment (bitfehler) [reply doc/rados/configuration/pool-pg.conf:19]: If I understand correctly, you are referring to the rule that lower numbers should be spelled out in sentences, right? Changed two instances of that and some other minor consistency issues in that comment, let me know if that's not what you wanted

**Review by bitfehler — COMMENTED** [review]
Thanks for all the feedback! I hope I accommodated it all, let me know if I missed something or misunderstood something.

> I encourage you to add a concise caveat to the description of osd_pool_default_pgp_num.

Gave it a try (very terse), please scrutinize

File: doc/rados/configuration/pool-pg.conf
Comment (bitfehler) [reply doc/rados/configuration/pool-pg.conf]: :joy: removed it, just wanted to draw more attention to the autoscaling, but the rest of changes are sufficient for that, so all good

File: doc/rados/operations/pools.rst
Comment (bitfehler) [reply doc/rados/operations/pools.rst]: > Also, while you're here, on lines 94 and 105 please s/-/_/

That's the only thing I didn't do, it seemed inconsistent? In those places, `pg-num`/`pgp-num` refer to the arguments of the `pool create` command, and arguments seem to use hyphens throughout the entire doc?

Just guessing, though, happy to be overridden if you say so.

File: doc/install/manual-deployment.rst
Comment (anthonyeleven) [reply doc/install/manual-deployment.rst:223]: Before your odyssey I hadn't even known `osd_pool_default_pgp_num` existed.  I converted a great number of option embedded spaces a couple of years ago, but clearly missed some.  So I'm always looking out for stragglers.  Ceph will accept with spaces, underscores, or I think even hypens, but IMHO underscores facilitate cut/paste and grepping.

File: doc/rados/configuration/pool-pg.conf
Comment (anthonyeleven) [reply doc/rados/configuration/pool-pg.conf:19]: Yes, I think the rule is any number under 10 / ten but there are probably variants.  Yes this is nitpicky.

File: doc/rados/operations/pools.rst
Comment (anthonyeleven) [reply doc/rados/operations/pools.rst]: Ah, I missed that.  Nevermind ;)

**Comment by anthonyeleven** [comment]
> > I encourage you to add a concise caveat to the description of osd_pool_default_pgp_num.
> 
> Gave it a try (very terse), please scrutinize

LGTM, thanks!  I've had to restrain myself at times from writing War+Peace in those descriptions, your wording conveys the idea appropriately for the context.

**Review by anthonyeleven — APPROVED** [review]

**Comment by anthonyeleven** [comment]
The failing test here seems unrelated

>stderr: fatal: detected dubious ownership in repository at '/home/jenkins-build/build/workspace/ceph-pull-requests-arm64'

@zdover23 since I butted my way into this PR, I'd be delighted if you could give it the once-over before merging.

**Comment by bitfehler** [comment]
gentle ping...

**Comment by zdover23** [comment]
@anthonyeleven: you beat me to it by less than five minutes!

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/51125 - Reef backport

Only fatigue and doubt about the applicability of this commit to Quincy and Pacific prevent my merging it to Quincy and Pacific. I will return to this.

**Pull request merged**
